### ISSUE TITLE
Improve paths in assets builder

### ISSUE THREAD

**Issue by boozook** [boozook/playdate#357]
Because there is infinite troubles with windows' paths.
Also just to ensure that all going well on each step.

- [ ] use something like `relative-path` or `typed-path`
- [ ] write more tests

**Comment by notpeter** [comment]
Yeah, compiling the stock hello world example throws this error on windows

```
C:\Users\peter\code\pd-hello>cargo +nightly playdate run
error: The filename, directory name, or volume label syntax is incorrect. (os error 123), caused when planning assets for pd-hello v0.1.0 (C:\Users\peter\code\pd-hello).
error: The filename, directory name, or volume label syntax is incorrect. (os error 123)
```

This error is caused by these lines in the generated Cargo.toml:
```
[package.metadata.playdate.assets]
"img/system/" = "${PLAYDATE_SDK_PATH}/Examples/Game Template/Source/SystemAssets/*.png"
"sfx/jump.wav" = "${PLAYDATE_SDK_PATH}/Examples/Level 1-1/Source/sfx/jump.wav"
```

It works if you hard code a windows path on the RHS so I assume it has something to do with the env var interpolation or something related to #352.

**Comment by boozook** [comment]
> `"img/system/" = "${PLAYDATE_SDK_PATH}/Examples/Game Template/Source/SystemAssets/*.png"`

Is's a glob in the path. There is blocking wax's issue https://github.com/olson-sean-k/wax/issues/34 , but I'll try to workaround it translating paths to normal slash `/` and escaping `:` in volume name - to make path more look like valid pattern.

**Comment by boozook** [comment]
Also, seems to current implementation works normally if run in the bash-like shell on windows, but not in powershell.

**Comment by bremco** [comment]
Hi! I just ran into something similar in bash (in Linux Manjaro) -- I had to remove the `{` and `}` from both lines mentioned above, otherwise I got:

~~~(bash)
[remco@deck pd-hello]$ cargo playdate run
error: malformed glob expression: adjacent component boundaries `/` or `**`, caused when planning assets for pd-hello v0.1.0 (/home/remco/code/pd-hello)
error: malformed glob expression: adjacent component boundaries `/` or `**`
~~~

So it seems it's not just a Windows problem.

**Comment by boozook** [comment]
@bremco Are you one hundred percent sure that the environment variables you specified are set? This may be if the variable is missed, because I removed the panic for the missed variables and instead of panic just left the path as it is. If I remember correctly. I know that it’s not good, it’s the point to change.

**Comment by bremco** [comment]
This is the one you mean right?

~~~(bash)
[remco@deck pd-hello]$ echo $PLAYDATE_SDK_PATH
/home/remco/opt/PlaydateSDK-2.6.1/
~~~

I'm only getting started with Rust, but I've been developing for the Playdate for a while. I always take care to set that environment variable, otherwise my own scripts would also fail :-)

In any case, everything (well, at least the basic example, both in the sim and on the device) works properly if I just remove `{}`s (at least in my case).

**Comment by boozook** [comment]
> This is the one you mean right?

Yes, exactly.

> works properly if I just remove `{}`s 

That's strange because [resolver with regex requires][regex] curly braces - `(${([^}]+)})`. And [there is default as-is string if can't resolve](https://github.com/boozook/playdate/blob/0786179185c94474a68297747c63cc1f3f42c570/support/build/src/assets/resolver.rs#L152).

Okay, anyway that resolver should be rewritten entirely, as I planned it. Thank you, I'll look at it soon.

[regex]: https://github.com/boozook/playdate/blob/0786179185c94474a68297747c63cc1f3f42c570/support/build/src/assets/resolver.rs#L116C41-L116C59
